src/protocol.rs

```diff
@@ -450,6 +466,50 @@ mod tests {
     use std::collections::HashMap;
     use tracing::debug;
 
+    #[test]
+    fn participant_config_has_more_than_one_participant() {
+        let result = std::panic::catch_unwind(|| {
+            let mut rng = init_testing();
+            let QUORUM_SIZE = 1;
+            let _p = ParticipantConfig::random_quorum(QUORUM_SIZE, &mut rng);
+        });
+
+        match result {
+            Ok(_) => {
+                println!("No panic occurred.");
+            }
+            Err(panic) => {
+                if let Some(s) = panic.downcast_ref::<String>() {
+                    println!("Panic occurred: {s}");
+                    assert_eq!(s, "Participant Config Size Error");
+                }
+            }
+        }
+    }
+
+    #[test]
+    fn participant_does_not_have_empty_set_of_other_participants() {
+        let result = std::panic::catch_unwind(|| {
+            let mut rng = init_testing();
+            let QUORUM_SIZE = 1;
+            let config = ParticipantConfig::random(QUORUM_SIZE, &mut rng);
```

Comment:
This still needs to be addressed: create a bad `ParticipantConfig` from scratch instead of using the `random` function. This isn't testing the `from_config` function at all right now (you can see because the test will still pass even if you comment out L517).

I might have function names wrong here, but like this:
```rust
let config = ParticipantConfig {
    self: ParticipantId::random(),
    other_ids: vec![],
}
```

This will also let you take out the panic unwinding code. This is supposed to test an external-facing function (`from_config`) which means we want to deal in recoverable errors, not unrecoverable ones.